lms/djangoapps/course_api/blocks/tests/test_api.py

```diff
@@ -77,3 +75,17 @@ def test_sub_structure(self):
                 self.assertIn(unicode(block.location), blocks['blocks'])
             else:
                 self.assertNotIn(unicode(block.location), blocks['blocks'])
+
+    def test_filtering_by_block_types(self):
+        sequential_block = self.store.get_item(self.course.id.make_usage_key('sequential', 'sequential_y1'))
+
+        block_types = ['problem']
+        blocks = get_blocks(self.request, sequential_block.location, self.user, block_types=block_types)
+
+        for block_type, block_name in (
+                ('problem', 'problem_y1a_1'),
+                ('problem', 'problem_y1a_2'),
+                ('problem', 'problem_y1a_3'),
+        ):
+            block = self.store.get_item(self.course.id.make_usage_key(block_type, block_name))
+            self.assertIn(unicode(block.location), blocks['blocks'])
```

Comment:
I think this test should show (through code or comment) that the source data has blocks with other block_types in addition to problem types.  Then, the final assertion should loop through all returned blocks to show that they are all problems.  As written, it feels like this would bass even if the block_types parameter was a no-op, unless I'm missing something.